lib/bsdlib/bsd_os.c

```diff
@@ -183,12 +336,23 @@ ISR_DIRECT_DECLARE(ipc_proxy_irq_handler)
 	IPC_IRQHandler();
 	ISR_DIRECT_PM(); /* PM done after servicing interrupt for best latency
 			  */
+
 	return 1; /* We should check if scheduling decision should be made */
 }
 
 ISR_DIRECT_DECLARE(rpc_proxy_irq_handler)
 {
+	atomic_inc(&rpc_event_cnt);
```

Comment:
Could the counter wrapping cause any issues? Not likely to happen soon, though..

Reply:
I think it should not, there is no direct comparison (other than `!=`) of this value in the code, so overflow should not be the problem.

Reply:
Right, and now I see that the calculation of thread age happens with IRQs locked.